miso-web/src/main/webapp/scripts/freezer_ajax.js

```diff
@@ -46,22 +54,40 @@
       save(!freezerJson || !freezerJson.id);
     });
   };
-  
-  Freezer.addRoom = function () {
-    Freezer.addRoomWithCallback(function() { Utils.showOkDialog('Room created', []); });
-  },
-  Freezer.addRoomWithCallback = function (callback) {
+
+  Freezer.validateAndSaveComponent = function() {
+    Validate.cleanFields('#freezerComponent-form');
+    Validate.clearErrors('#freezerComponent-form');
+
+    $('#storageComponentBarcode').attr('class', 'form-control');
+    $('#storageComponentBarcode').attr('data-parsley-maxlength', '255');
+    $('#storageComponentBarcode').attr('data-parsley-pattern', Utils.validation.sanitizeRegex);
+    $('#storageComponentBarcode').attr('data-parsley-errors-container', '#storageComponentBarcodeError');
+
+    $('#freezerComponent-form').parsley();
+    $('#freezerComponent-form').parsley().validate();
+
+    Validate.updateWarningOrSubmit('#freezerComponent-form', null, function() {
+      saveComponent();
+    });
+  };
+
+  Freezer.addRoom = function() {
+    Freezer.addRoomWithCallback(function() {
+      Utils.showOkDialog('Room created', []);
+    });
+  }, Freezer.addRoomWithCallback = function(callback) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove use of this comma operator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS878)